src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/GenericWhitespaceCheck.java

```diff
@@ -282,8 +282,7 @@ private static boolean isGenericBeforeCtor(DetailAST ast) {
      * @return true if generic before a method ref
      */
     private static boolean isGenericBeforeMethod(DetailAST ast) {
-        return ast.getParent().getParent().getType() == TokenTypes.DOT
-                && ast.getParent().getParent().getParent().getType() == TokenTypes.METHOD_CALL
+        return ast.getParent().getParent().getParent().getType() == TokenTypes.METHOD_CALL
```

Comment:
can you explain why root node introduction results in removal of extra expression ?

Reply:
`ast.getParent().getParent().getType() == TokenTypes.DOT` was really only guarding against NPE from `ast.getParent().getParent().getParent()` being null. Now that `ast.getParent().getParent().getParent()` can never be null, this line failed pitest.
`TYPE_PARAMETERS` will always have at least three parents now that we have `COMPILATION_UNIT`. 

This was the only line that covered the original condition:

```java

interface SupplierFunction<T> extends Map<List<T>, T> {}

```

Which results in (master):
```bash

ast type: GENERIC_END
ast parent type: TYPE_PARAMETERS
ast parent's parent type: INTERFACE_DEF
ast parent's parent's parent type: null

```

And (this PR):

```bash

ast type: GENERIC_END
ast parent type: TYPE_PARAMETERS
ast parent's parent type: INTERFACE_DEF
ast parent's parent's parent type: COMPILATION_UNIT

```